src/crimson/os/seastore/cache.cc

```diff
@@ -947,6 +947,16 @@ record_t Cache::prepare_record(Transaction &t)
     get_by_ext(efforts.retire_by_ext,
                i->get_type()).increment(i->get_length());
     retire_extent(i);
+    if (i->backend_type == device_type_t::RANDOM_BLOCK) {
+      paddr_t paddr = i->get_paddr();
+      ceph_assert(rbm);
+      blk_paddr_t rbm_addr =
+	convert_paddr_to_blk_paddr(paddr, rbm->get_block_size(),
+	  rbm->get_blocks_per_segment());
+      rbm->free_extent(t,
```

Comment:
This appears to be the only user of the rpm member.  I don't think this is a good place for this.  In general, there may be more than one RandomBlockDevice device.  Every other read in Cache deals with that possibility by routing the read through the ExtentReader interfaces.

What does free_extent do here?  NVMeManager::add_free_extent seem to just add an rbm_alloc_delta_t to Transaction::rbm_alloc_info_blocks.  The only thing the NVMeManager instance is actually used for is to translate the blk_paddr_t to a blk_id_t.  This process seems as though it *must* be identical for every RandomBlockManger implementation since it's constrained by the journal format and Transaction structures.  Thus:
1) Change rbm_alloc_delta_t to use an interval_set<rbm_offset_t> determined directly from the paddr_t.
2) Remove RandomBlockManager::free_extent and Cache::rbm entirely -- every implementation must behave the same way.